googleapi/googleapi.go

```diff
@@ -145,22 +145,40 @@ func CheckResponse(res *http.Response) error {
 	}
 	slurp, err := io.ReadAll(res.Body)
```

Comment:
@codyoss Did you ever notice that if `err` from `ReadAll` is non-nil, it is never used (lines 150-154, below)?

Reply:
I believe the intention of this function is to provide more context to the error in all cases, that is why it is never returned directly. We can at least always append some HTTP info.